underfs/cephfs/src/main/java/alluxio/underfs/cephfs/CephFSUnderFileSystem.java

```diff
@@ -0,0 +1,789 @@
+      RetryPolicy retryPolicy = new CountingRetry(MAX_TRY);
+      while (retryPolicy.attempt()) {
+        try {
+          return deleteInternal(path, options.isRecursive());
+        } catch (IOException e) {
+          LOG.warn("Retry count {} : {}", retryPolicy.getAttemptCount(), e.getMessage());
+          te = e;
+        }
+      }
+      throw te;
+    }
+    return false;
+  }
+
+  @Override
+  public boolean deleteExistingDirectory(String path) throws IOException {
+    return deleteDirectory(path);
+  }
+
+  @Override
+  public boolean deleteExistingDirectory(String path, DeleteOptions options) throws IOException {
+    return deleteDirectory(path, options);
+  }
+
+  @Override
+  public boolean deleteFile(String path) throws IOException {
+    path = stripPath(path);
+    if (isFile(path)) {
+      IOException te = null;
+      RetryPolicy retryPolicy = new CountingRetry(MAX_TRY);
+      while (retryPolicy.attempt()) {
+        try {
+          return deleteInternal(path, false);
+        } catch (IOException e) {
+          LOG.warn("Retry count {} : {}", retryPolicy.getAttemptCount(), e.getMessage());
+          te = e;
+        }
+      }
+      throw te;
+    }
+    return false;
+  }
+
+  @Override
+  public boolean deleteExistingFile(String path) throws IOException {
+    return deleteFile(path);
+  }
+
+  @Override
+  public boolean exists(String path) throws IOException {
+    path = stripPath(path);
+    try {
+      CephStat stat = new CephStat();
+      lstat(path, stat);
+      return true;
+    } catch (FileNotFoundException e) {
+      return false;
+    }
+  }
+
+  @Override
+  public long getBlockSizeByte(String path) throws IOException {
+    path = stripPath(path);
+    CephStat stat = new CephStat();
+    lstat(path, stat);
+
+    return stat.blksize;
+  }
+
+  @Override
+  public UfsDirectoryStatus getDirectoryStatus(String path) throws IOException {
+    path = stripPath(path);
+    CephStat stat = new CephStat();
+    lstat(path, stat);
+
+    return new UfsDirectoryStatus(path, "", "", (short) stat.mode);
+  }
+
+  @Override
+  public UfsDirectoryStatus getExistingDirectoryStatus(String path) throws IOException {
+    return getDirectoryStatus(path);
+  }
+
+  @Override
+  public UfsStatus getStatus(String path) throws IOException {
+    path = stripPath(path);
+    CephStat stat = new CephStat();
+    lstat(path, stat);
+    if (stat.isFile()) {
+      return getFileStatus(path);
+    } else if (stat.isDir()) {
+      return getDirectoryStatus(path);
+    }
+
+    throw new IOException("Failed to getStatus: " + path);
+  }
+
+  @Override
+  public UfsStatus getExistingStatus(String path) throws IOException {
+    return getStatus(path);
+  }
+
+  @Override
+  public List<String> getFileLocations(String path) throws IOException {
+    return null;
+  }
+
+  @Override
+  @Nullable
+  public List<String> getFileLocations(String path, FileLocationOptions options)
+      throws IOException {
+    return null;
```

Comment:
Not sure if any problem will be introduced by this null return value, please help make sure it's safe in workloads